src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/EdgeCasesTests.cs

```diff
@@ -0,0 +1,423 @@
+                + "96726216e986869eed578bda652855c85604a056201538ee56b6c4302b06092a864886f70d010701301406082a864886f70d"
+                + "030704083adadf63cd297a86800835edc437e31d0b70").HexToByteArray();
+
+            EnvelopedCms ecms = new EnvelopedCms();
+            ecms.Decode(encodedMessage);
+            RecipientInfo recipientInfo = ecms.RecipientInfos[0];
+            X509Certificate2Collection extraStore = null;
+            Assert.Throws<ArgumentNullException>(() => ecms.Decrypt(extraStore));
+            Assert.Throws<ArgumentNullException>(() => ecms.Decrypt(recipientInfo, extraStore));
+        }
+
+        [Fact]
+        public static void EnvelopedCmsDecryptWithoutMatchingCert()
+        {
+            // You don't have the private key? No message for you.
+
+            // This is the private key that "we don't have." We want to force it to load anyway, though, to trigger
+            // the "fail the test due to bad machine config" exception if someone left this cert in the MY store check. 
+            using (X509Certificate2 ignore = Certificates.RSAKeyTransfer1.TryGetCertificateWithPrivateKey())
+            { }
+
+            byte[] encodedMessage =
+                ("3082010c06092a864886f70d010703a081fe3081fb0201003181c83081c5020100302e301a311830160603550403130f5253"
+                + "414b65795472616e7366657231021031d935fb63e8cfab48a0bf7b397b67c0300d06092a864886f70d01010105000481805e"
+                + "bb2d08773594be9ec5d30c0707cf339f2b982a4f0797b74d520a0c973d668a9a6ad9d28066ef36e5b5620fef67f4d79ee50c"
+                + "25eb999f0c656548347d5676ac4b779f8fce2b87e6388fbe483bb0fcf78ab1f1ff29169600401fded7b2803a0bf96cc160c4"
+                + "96726216e986869eed578bda652855c85604a056201538ee56b6c4302b06092a864886f70d010701301406082a864886f70d"
+                + "030704083adadf63cd297a86800835edc437e31d0b70").HexToByteArray();
+
+            EnvelopedCms ecms = new EnvelopedCms();
+            ecms.Decode(encodedMessage);
+            RecipientInfo recipientInfo = ecms.RecipientInfos[0];
+            X509Certificate2Collection extraStore = new X509Certificate2Collection();
+            Assert.ThrowsAny<CryptographicException>(() => ecms.Decrypt(recipientInfo));
+            Assert.ThrowsAny<CryptographicException>(() => ecms.Decrypt(extraStore));
+            Assert.ThrowsAny<CryptographicException>(() => ecms.Decrypt(recipientInfo, extraStore));
+        }
+
+        [Fact]
+        public static void EnvelopedCmsDecryptWithoutMatchingCertSki()
+        {
+            // You don't have the private key? No message for you.
+
+            // This is the private key that "we don't have." We want to force it to load anyway, though, to trigger
+            // the "fail the test due to bad machine config" exception if someone left this cert in the MY store check. 
+            using (X509Certificate2 ignore = Certificates.RSAKeyTransfer1.TryGetCertificateWithPrivateKey())
```

Comment:
If I were writing this from scratch I'd probably encode this as a [ConditionalFact] of checking the store (so if you're in that state it says SKIP instead of FAIL).  But, it's not a big deal.